### PR TITLE
issue 607 sort movies - part 1

### PR DESCRIPTION
**Pull request by Maxstgt**
this is part 1 of the issue 607. I need to split it up because I had to rename the movie list guard and the movie active guard, because they are working on the organization movieID property and the naming was not fitting. I did some setup for sort behavior of the search movie page. But this will update when @ioaNikas has worked on the akita-fire lib.

### CODE REVIEW

File: apps/catalog/marketplace/marketplace/src/app/guards/catalog-movie-list.guard.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Router } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { Query, StateListGuard, FireQuery } from '@blockframes/utils';
+import { Movie, MovieStore } from '@blockframes/movie';
+
+export const catalogMovieListQuery = (): Query<Movie> => ({
+  path: `movies`
+});
+@Injectable({ providedIn: 'root' })
```
Comment (pldespaigne) [inline apps/catalog/marketplace/marketplace/src/app/guards/catalog-movie-list.guard.ts]: break line please

File: apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.html [hunk 2]
```diff
@@ -177,13 +177,31 @@ <h6>You can choose multiple certifications</h6>
       <div class="movies" fxLayout="row" fxLayoutAlign="start center" fxLayoutGap="10px">
         <h3>Films</h3>
         <mat-divider></mat-divider>
+        <button mat-icon-button (click)="cardOrList = !cardOrList" *ngIf="!cardOrList">
+          <mat-icon>list</mat-icon>
+        </button>
+        <button mat-icon-button (click)="cardOrList = !cardOrList" *ngIf="cardOrList">
```
Comment (pldespaigne) [inline apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.html]: those two buttons does the same thing, would it not be better to have one button that change its icon ?

Comment (pldespaigne) [reply apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.html]: also with the name `cardOrList` it is hard to tell if it's a card when the value is true or when the value is false

File: apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts [hunk 3]
```diff
@@ -30,11 +34,21 @@ export class MyErrorStateMatcher implements ErrorStateMatcher {
...
   public movieSearchResults$: Observable<Movie[]>;
+
+  /* Instance of the search form */
   public filterForm = new CatalogSearchForm();
+
+  /* Array of sorting options */
+  public sortOptions: string[] = ['All films', 'Title', 'Director', 'Production Year'];
+
+  /* Flag to indicate either the movies should be presented as a card or a list */
+  public cardOrList: boolean;
```
Comment (pldespaigne) [inline apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts]: prefer a name like `isCard` or `isList`

File: libs/movie/src/lib/movie/components/display-list/display-list.component.scss [hunk 4]
```diff
@@ -0,0 +1,10 @@
+@import 'variables';
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-list/display-list.component.scss]: unused import

File: libs/movie/src/lib/movie/components/display-list/display-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Movie } from './../../+state/movie.model';
+import { Component, Input } from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+
+@Component({
+  selector: '[movie] movie-display-list',
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-list/display-list.component.ts]: it's just a question : What does `[movie]` means in the selector ?

File: libs/movie/src/lib/movie/components/display-list/display-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  styleUrls: ['./display-list.component.scss']
+})
+export class MovieDisplayListComponent {
+  public displayedColumns: string[] = ['picture', 'title', 'director', 'productionYear'];
+  public dataSource;
+
+  @Input()
+  set movie(movie: Movie[]) {
+    this.dataSource = new MatTableDataSource(movie);
+    console.log(this.dataSource);
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-list/display-list.component.ts]: remove `console.log`

File: libs/movie/src/lib/movie/components/display-list/display-list.component.ts (hunk 5 above)
Comment (Maxstgt) [reply libs/movie/src/lib/movie/components/display-list/display-list.component.ts]: you force your parent component to give you an input with that name

**Pull request merged**